vcd/resource_vcd_lb_server_pool.go

```diff
@@ -0,0 +1,344 @@
+	}
+
+	return setLBPoolData(d, updatedLBPool)
+}
+
+func resourceVcdLBServerPoolDelete(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	err = edgeGateway.DeleteLBServerPool(&types.LBPool{ID: d.Id()})
+	if err != nil {
+		return fmt.Errorf("error deleting load balancer server pool: %s", err)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func resourceVcdLBServerPoolImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	resourceURI := strings.Split(d.Id(), ".")
+	if len(resourceURI) != 4 {
+		return nil, fmt.Errorf("resource name must be specified as org.VDC.edge-gw.lb-server-pool")
+	}
+	orgName, vdcName, edgeName, poolName := resourceURI[0], resourceURI[1], resourceURI[2], resourceURI[3]
+
+	vcdClient := meta.(*VCDClient)
+	edgeGateway, err := vcdClient.GetEdgeGateway(orgName, vdcName, edgeName)
+	if err != nil {
+		return nil, fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	readLBPool, err := edgeGateway.ReadLBServerPool(&types.LBPool{Name: poolName})
+	if err != nil {
+		return []*schema.ResourceData{}, fmt.Errorf("unable to find load balancer server pool with name %s: %s", d.Id(), err)
+	}
+
+	d.Set("org", orgName)
+	d.Set("vdc", vdcName)
+	d.Set("edge_gateway", edgeName)
+	d.Set("name", poolName)
+
+	d.SetId(readLBPool.ID)
+	return []*schema.ResourceData{d}, nil
+}
+
+// getLBPoolType converts schema.ResourceData to *types.LBPool and is useful
+// for creating API requests
+func getLBPoolType(d *schema.ResourceData) (*types.LBPool, error) {
+	lbPool := &types.LBPool{
+		Name:                d.Get("name").(string),
+		Description:         d.Get("description").(string),
+		Algorithm:           d.Get("algorithm").(string),
+		MonitorId:           d.Get("monitor_id").(string),
+		Transparent:         d.Get("enable_transparency").(bool),
+		AlgorithmParameters: d.Get("algorithm_parameters").(string),
+	}
+
+	members, err := getLBPoolMembersType(d)
+	if err != nil {
+		return nil, err
+	}
+	lbPool.Members = members
+
+	return lbPool, nil
+}
+
+// getLBPoolMembersType converts schema.ResourceData to *types.LBPoolMembers and is useful
+// for creating API requests
+func getLBPoolMembersType(d *schema.ResourceData) (types.LBPoolMembers, error) {
+	var lbPoolMembers types.LBPoolMembers
+
+	members := d.Get("member").([]interface{})
+	for _, memberInterface := range members {
+		var memberConfig types.LBPoolMember
+		member := memberInterface.(map[string]interface{})
+
+		// If we have IDs - then we must insert them for update. Otherwise the update may get mixed
+		if member["id"].(string) != "" {
+			memberConfig.ID = member["id"].(string)
+		}
+
+		memberConfig.Name = member["name"].(string)
+		memberConfig.IpAddress = member["ip_address"].(string)
```

Comment:
It won't panic  if `ip_address` is not defined?

Reply:
Very goot spot! I made a mistake in requirements. It is mandatory.

Reply:
Fixed in `ip_address` definition